index.js

```diff
@@ -306,6 +306,27 @@ exports.isValidPrivate = function (privateKey) {
 }
 
 /**
+ * Checks if the public key satisfies the rules of the curve secp256k1
+ * and the requirements of Ethereum.
+ * @method isValidPublic
+ * @param {Buffer} publicKey The two points of an uncompressed key, unless sanitize is enabled
+ * @param {Boolean} [sanitize=false] Accept public keys in other formats
+ * @return {Boolean}
+ */
+exports.isValidPublic = function (publicKey, sanitize) {
+  if (publicKey.length === 64) {
+    // Convert to SEC1 for secp256k1
+    return secp256k1.publicKeyVerify(Buffer.concat([ new Buffer('04', 'hex'), publicKey ]))
```

Comment:
The Buffer.copy is nice, but don't want to pollute the context with random data. Do you think moving the whole into the if would still be faster than Buffer.contact? I think it would.

So 0x06 means it is odd and 0x07 means it is even (or the other way around?); while 0x04 doesn't do that check?